lib/linked_list.rb

```diff
@@ -18,122 +18,289 @@ def initialize
 
     # method to add a new node with the specific data value in the linked list
     # insert the new node at the beginning of the linked list
-    # Time Complexity: ?
-    # Space Complexity: ?
+    # Time Complexity: O(1) => because we don't need to traverse 
+    # Space Complexity: O(1) => because we are not using any extra space
     def add_first(value)
-      raise NotImplementedError
+      if @head.nil?
+        @head = Node.new(value)
+      else 
+        @head = Node.new(value, @head)

```

Comment:
Can be simplified to 

```suggestion
     @head = Node.new(value, @head)
```